packages/teams-js/src/public/marketplace.ts

```diff
@@ -0,0 +1,211 @@
+  readonly intent: Intent;
+  /** Locale for the user */
+  readonly locale: Locale;
+  /** The status of the cart. */
+  status: CartStatus;
+  /** Currency code for the cart item price. */
+  readonly currency: Currency;
+  /** ISO 8601 timestamp string, indicates when the cart is created. */
+  readonly createdAt: string;
+  /** ISO 8601 timestamp string, indicates when the cart is updated. */
+  updatedAt: string;
+}
+
+/**
+ * Represents the basic cart item information.
+ *
+ * @beta
+ */
+interface Item {
+  /** The id of the cart item. */
+  readonly id: string;
+  /** The display name of the cart item. */
+  name: string;
+  /** The quantity of the cart item. */
+  quantity: number;
+  /** The price of the single cart item. */
+  price: number;
+  /** The thumbnail imageURL of the cart item. */
+  readonly imageURL?: string;
+}
+
+/**
+ * Represents the cart item that could have accessories
+ *
+ * @beta
+ */
+export interface CartItem extends Item {
+  /** Accessories to the item if existing. */
+  readonly accessories?: Item[];
+}
+
+/**
+ * Represents the persona creating the cart.
+ *
+ * @beta
+ */
+export enum Intent {
+  /** The cart is created by admin of an organization. */
+  AdminUser = 'AdminUser',
+  /** The cart is created by end user of an organization. */
+  EndUser = 'EndUser',
+}
+
+/**
+ * Represents the status of the cart.
+ *
+ * @beta
+ */
+export enum CartStatus {
+  /** Cart is created but not checked out yet. */
+  Open = 'Open',
+  /** Cart is checked out but not completed yet. */
+  Processing = 'Processing',
+  /**
+   * Indicate checking out is completed and the host should
+   * Response a new cart in the next getCart call.
+   */
+  Processed = 'Processed',
+  /**
+   * Indicate checking out is failed and the host should
+   * Response a new cart in the next getCart call.
+   */
+  Error = 'Error',
+}
+/**
+ * Represents the parameters to update the cart status.
+ *
+ * @beta
+ */
+export interface UpdateCartStatusParams {
+  /** Status of the cart. */
+  cartStatus: CartStatus;
+  /** Extra info to the status. */
+  statusInfo?: string;
+}
+
+/**
+ * Namespace for a vendor app to support a checkout flow by interacting with the marketplace cart in the host.
+ */
+export namespace marketplace {
+  /**
+   * Get the cart object owned by the host to checkout.
+   *
+   * @beta
+   */
+  export function getCart(): Promise<Cart> {
+    return new Promise<Cart>((resolve) => {
+      ensureInitialized(runtime, FrameContexts.content);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+      resolve(sendAndHandleSdkError('marketplace.getCart'));
+    });
+  }
+  /**
+   * Add or update cart items in the cart owned by the host.
+   *
+   * @param cartItems - A list of cart items, if item id exists, overwrite the item, otherwise add new items to cart.
+   *
+   * @beta
+   */
+  export function addOrUpdateCartItems(cartItems: CartItem[]): Promise<void> {
```

Comment:
It seems like you would want to pass up the cart Id at least here in case you need that in the host. May also be worth considering whether this should return Promise<Cart> instead of Promise<void>, in case developers want to render the updated card state after calling this method

Reply:
Is the array of items in here intended to just be the changed or added items? What happens if a value is passed in that is the same as the current value in the cart?